sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/SclService.java

```diff
@@ -560,73 +559,65 @@ public static void removeAllControlBlocksAndDatasetsAndExtRefSrcBindings(final S
-     * @return SclReport Object that contain SCL file and set of errors
+     * @return list of encountered errors
      */
-    public static SclReport analyzeDataGroups(SCL scd) {
+    public static List<SclReportItem> analyzeDataGroups(SCL scd) {
         SclRootAdapter sclRootAdapter = new SclRootAdapter(scd);
-        List<SclReportItem> sclReportItems = sclRootAdapter.streamIEDAdapters()
+        return sclRootAdapter.streamIEDAdapters()
                 .map(iedAdapter -> {
                     List<SclReportItem> list = new ArrayList<>();
                     list.addAll(iedAdapter.checkDataGroupCoherence());
                     list.addAll(iedAdapter.checkBindingDataGroupCoherence());
                     return list;
-                }).flatMap(Collection::stream).toList();
-
-        return new SclReport(sclRootAdapter, sclReportItems);
+                }).flatMap(Collection::stream).collect(Collectors.toList());
     }
 
     /**
      * Update DAIs of DO InRef in all LN0 of the SCD using matching ExtRef information.
      *
      * @param scd SCL file for which DOs InRef should be updated with matching ExtRef information
-     * @return SclReport Object that contain SCL file and set of errors
+     * @return list of encountered errors
      */
-    public static SclReport updateDoInRef(SCL scd) {
+    public static List<SclReportItem> updateDoInRef(SCL scd) {
         SclRootAdapter sclRootAdapter = new SclRootAdapter(scd);
-        List<SclReportItem> sclReportItems = sclRootAdapter.streamIEDAdapters()
+        return sclRootAdapter.streamIEDAdapters()
                 .flatMap(IEDAdapter::streamLDeviceAdapters)
                 .map(LDeviceAdapter::getLN0Adapter)
                 .map(LN0Adapter::updateDoInRef)
                 .flatMap(List::stream)
-                .toList();
-        return new SclReport(sclRootAdapter, sclReportItems);
+                .collect(Collectors.toList());
     }
 
     /**
      * Update and/or create Monitoring LNs (LSVS and LGOS) for bound GOOSE and SMV Control Blocks
      *
      * @param scd SCL file for which  LNs (LSVS and LGOS) should be updated and/or created in each LDevice LDSUIED with matching ExtRef information
-     * @return SclReport Object that contain SCL file and set of errors
+     * @return list of encountered errors
      */
-    public static SclReport manageMonitoringLns(SCL scd) {
+    public static List<SclReportItem> manageMonitoringLns(SCL scd) {
         SclRootAdapter sclRootAdapter = new SclRootAdapter(scd);
-        List<SclReportItem> sclReportItems = sclRootAdapter.streamIEDAdapters()
+        return sclRootAdapter.streamIEDAdapters()
                 .filter(iedAdapter -> !iedAdapter.getName().contains(IED_TEST_NAME))
                 .map(IEDAdapter::manageMonitoringLns)
                 .flatMap(List::stream)
-                .toList();
-        return new SclReport(sclRootAdapter, sclReportItems);
+                .collect(Collectors.toList());
```

Comment:
Use `.toList()`. Caller should not assume that the returned List is modifiable.

Reply:
done

Reply:
```suggestion
                .toList();
```